table/index.go

```diff
@@ -40,7 +40,7 @@ type Index interface {
 	// Create supports insert into statement.
 	Create(ctx sessionctx.Context, rm kv.RetrieverMutator, indexedValues []types.Datum, h int64, opts ...*CreateIdxOpt) (int64, error)
 	// Delete supports delete from statement.
-	Delete(sc *stmtctx.StatementContext, m kv.Mutator, indexedValues []types.Datum, h int64) error
+	Delete(sc *stmtctx.StatementContext, m kv.Mutator, indexedValues []types.Datum, h int64, ss kv.Transaction) error
```

Comment:
I think it's better to keep this method unchanged and let the caller explicitly call `txn.SetAssertion` if needed.
Passing a `Transaction` or nil to indicate if assertion is needed is not straightforward.

Reply:
> it's better to keep this method unchanged

I agree, but the caller has to know the key, `SetAssertion(key, xxx)`, the key is obtained inside the `Delete` function. @coocood